viewer/packages/sector-parser/src/GltfSectorParser.ts

```diff
@@ -2,224 +2,289 @@
-    primitive: Primitive,
-    bufferGeometry: THREE.BufferGeometry | THREE.InstancedBufferGeometry
-  ) {
-    const treeIndexAttribute = primitive.getAttribute('_treeIndex')!;
-    const colorAttribute = primitive.getAttribute('COLOR_0')!;
+    assert(uniqueBufferViews.length === 1, 'Unexpected number of buffer views');
 
-    bufferGeometry.setAttribute(
-      'color',
-      new THREE.BufferAttribute(colorAttribute.getArray()!, colorAttribute.getElementSize())
-    );
+    const bufferView = json.bufferViews[uniqueBufferViews[0]];
+    bufferView.byteOffset = bufferView.byteOffset ?? 0;
+
+    const offsetToBinChunk = glbHeaderData.byteOffsetToBinContent;
 
-    bufferGeometry.setAttribute(
-      'a_treeIndex',
-      new THREE.BufferAttribute(treeIndexAttribute.getArray()!, treeIndexAttribute.getElementSize())
+    const componentTypes = Object.values(attributes).map(accessorId => json.accessors[accessorId].componentType);
+
+    const typedArrayMap: { [key: string]: T } = this.getUniqueComponentViews<T>(
+      componentTypes,
+      data,
+      offsetToBinChunk,
+      bufferView,
+      bufferType
     );
+
+    this.setAttributes<T>(attributes, json, typedArrayMap, transformAttributeName, bufferGeometry);
   }
 
-  private setMeshAttributes(
-    primitive: Primitive,
-    bufferGeometry: THREE.BufferGeometry | THREE.InstancedBufferGeometry
+  private setAttributes<T extends THREE.InterleavedBuffer>(
+    attributes: { [key: string]: number },
+    json: GltfJson,
+    typedArrayMap: { [key: string]: T },
+    transformAttributeName: (attributeName: string) => string,
+    bufferGeometry: THREE.InstancedBufferGeometry | THREE.BufferGeometry
   ) {
-    const indicesAccessor = primitive.getIndices()!;
-    const positionAttribute = primitive.getAttribute('POSITION')!;
+    Object.keys(attributes).forEach(attributeName => {
+      const accessor = json.accessors[attributes[attributeName]];
 
-    bufferGeometry.setAttribute(
-      'position',
-      new THREE.BufferAttribute(positionAttribute.getArray()!, positionAttribute.getElementSize())
-    );
-    bufferGeometry.setIndex(new THREE.BufferAttribute(indicesAccessor.getArray()!, indicesAccessor.getElementSize()));
+      const interleavedBuffer = typedArrayMap[accessor.componentType];
+      const size = this.COLLECTION_TYPE_SIZES.get(accessor.type);
+
+      assert(size !== undefined);
+
+      const elementType = this.DATA_TYPE_BYTE_SIZES.get(accessor.componentType);
+
+      assert(elementType !== undefined);
+
+      const elementSize = elementType.BYTES_PER_ELEMENT;
+
+      assert(elementSize !== undefined);
+
+      const interleavedBufferAttribute = new THREE.InterleavedBufferAttribute(
+        interleavedBuffer,
+        size,
+        (accessor.byteOffset ?? 0) / elementSize
+      );
+      const transformedAttributeName = transformAttributeName(attributeName);
+      bufferGeometry.setAttribute(transformedAttributeName, interleavedBufferAttribute);
```

Comment:
```suggestion
      const interleavedBuffer = typedArrayMap[accessor.componentType];
      const size = this.COLLECTION_TYPE_SIZES.get(accessor.type);
      assert(size !== undefined);
      const elementType = this.DATA_TYPE_BYTE_SIZES.get(accessor.componentType);
      assert(elementType !== undefined);
      const elementSize = elementType.BYTES_PER_ELEMENT;
      assert(elementSize !== undefined);

      const interleavedBufferAttribute = new THREE.InterleavedBufferAttribute(
        interleavedBuffer,
        size,
        (accessor.byteOffset ?? 0) / elementSize
      );
      const transformedAttributeName = transformAttributeName(attributeName);
      bufferGeometry.setAttribute(transformedAttributeName, interleavedBufferAttribute);
```
To me, it doesn't make the code more readable with all of these blank lines, but just a suggestion

Reply:
I don't agree with this, and prefer the space when reading. Not obsessed with this though so can change if you are very opinionated about this.